MySQL.php

```diff
@@ -724,8 +745,14 @@ public function mysql_set_charset($charset, $link = false)
      */
     public function mysql_insert_id($link = false)
     {
-        $link = $this->_getLastLink($link);
-        return $this->_instances[$link]->lastInsertId();
+        if(!$link instanceof PDOStatement) {
```

Comment:
I see what you are trying to do here. But `_getLastLink` already takes care of this for you so you don't have to add another duplicate check, The same is true for all the other methods that use that same call.